pkg/sql/insert_fast_path.go

```diff
@@ -282,6 +283,14 @@ func (n *insertFastPathNode) BatchedNext(params runParams) (bool, error) {
 			var err error
 			inputRow[col], err = typedExpr.Eval(params.EvalContext())
 			if err != nil {
+				// Intercept parse error due to ALTER COLUMN TYPE schema change.
+				if n.run.insertRun.insertCols[col].AlterColumnTypeDropColumn {
```

Comment:
I think that you also want to put this error catch in the insert non-fast path node, and the updater.  It would be good to wrap this in a function for easy reuse